### DISCUSSION TITLE
how to support ext4 filesystem?

### DISCUSSION THREAD

**Discussion by WhatDaniel** [nlitsme/extfstools#16] [Q&A] [answered]
now, i use ext2rd to extract files from image which is ext4 filesystem, but sometimes ext2rd could not get file inode, this puzzles me a lot, i want get some suggestions to sovle this problem, please check me!
environment: ubuntu18 x64

**Comment by nlitsme** [comment, marked as answer]
my tool currently does not know how to handle ext4 filesystems. so it will only show you the parts that deal only with ext2 compatible data.

the solution: look at how ext4 extends ext2,  and implement that.

**Reply by WhatDaniel** [reply]
more: i did a test. if i got the ext4 image from other system, and when i use ext2rd to extract files from it, maybe i just only get some not all. but if i mount it on some place and umount it, most likely i could get all files using ext2rd, why does this happen and what affects the extraction?

**Reply by nlitsme** [reply]
when you try to extract files from an ext4 filesystem using ext2rd,  you will most likely end up with incomplete files.

I did not look into what ext4 details are missing.   I wrote this tool a long time ago, when I needed to inspect ext2 filesystems on my then windows laptop,  nowadays I have a linux laptop, so my need for improving this tool is minimal.

**Reply by WhatDaniel** [reply]
OK，thank you so much.  Maybe i could extend this tool to support ext4 filesystem, thanks again!
